### PR TITLE
Update goimports-reviser to version that properly ignores generated files

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

**What this PR does / why we need it**:
Update goimports-reviser to version that properly ignores generated files
Ref: https://github.com/incu6us/goimports-reviser/pull/95

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature dependency
The `goimports-reviser` is updated to a version that properly ignores generated files.
```

### CODE REVIEW

File: hack/tools.mk [hunk 1]
```diff
@@ -59,7 +59,7 @@ YQ                         := $(TOOLS_BIN_DIR)/yq
 DOCFORGE_VERSION ?= v0.33.0
 GOLANGCI_LINT_VERSION ?= v1.50.1
 GO_APIDIFF_VERSION ?= v0.5.0
-GOIMPORTSREVISER_VERSION ?= v3.3.0
+GOIMPORTSREVISER_VERSION ?= 32c80678d5d73a50b6966f06b346de58b1d018f1
```
Comment (ialidzhikov) [inline hack/tools.mk:63]:
```suggestion
# TODO(vpnachev): Update goimports-reviser to v3.4.0 when there is a release including https://github.com/incu6us/goimports-reviser/pull/95.
GOIMPORTSREVISER_VERSION ?= 32c80678d5d73a50b6966f06b346de58b1d018f1
```

Comment (ialidzhikov) [inline hack/tools.mk:62]: With this change I believe we no longer need to call `make format` in `make generate`. This was introduced in https://github.com/gardener/gardener/pull/7349.
If that is the case, we also need to add a release note as currently all the extensions have this `make format` call in their `make generate`.

**Review by ialidzhikov — COMMENTED** [review]
Nice! I was also not happy that `make format` (goimports-reviser) is messing with generated files ~~and that it was workarounded by calling `make format` in `make generate`~~ (the `make format` was present in `make generate` even before the goimports-reviser)

**Review by ialidzhikov — COMMENTED** [review]
I now double-checked locally. Works like a charm!

/lgtm

**Review by acumino — COMMENTED** [review]
Thanks!
/approve

**Pull request merged**